src/main.rs

```diff
@@ -35,16 +43,26 @@ async fn main() -> anyhow::Result<()> {
     dotenv().ok();
     let opts: Opts = Opts::parse();
 
-    let pool = sqlx::PgPool::connect(&env::var("DATABASE_URL")?).await?;
+    let s3_config = aws_sdk_s3::config::Builder::from(&opts.lake_aws_sdk_config()).build();
+    let config_builder = near_lake_framework::LakeConfigBuilder::default().s3_config(s3_config);
 
-    let config = near_lake_framework::LakeConfigBuilder::default()
-        .s3_bucket_name(opts.s3_bucket_name)
-        .s3_region_name(opts.s3_region_name)
+    let config = config_builder
+        .mainnet()
         .start_block_height(opts.start_block_height)
-        .blocks_preload_pool_size(100)
         .build()?;
+    let pool = sqlx::PgPool::connect(&env::var("DATABASE_URL")?).await?;
 
-    let _writer_guard = init_tracing();
+    let env_filter = EnvFilter::new("near_lake_framework=info,indexer_events=info");
+    let _subscriber = tracing_utils::init_tracing(env_filter).await;
+
+    task::spawn_blocking(move || {
```

Comment:
I am still getting familiar with async rust. It does not seem to have any performance issues, but potentially a place to make improvements using a non blocking thread. Tasks seem to very lightweight according to  [docs](https://docs.rs/tokio/1.22.0/tokio/task/index.html)